packages/language-service/src/lib-new/lang-service-context.ts

```diff
@@ -160,7 +170,11 @@ export class LangServiceContext {
                     (last) => !!last.length
                 );
                 if (upperNestingSelectors) {
-                    // ToDo: handle multiple selector intersection type
+                    // add nest level multiple selector inferred intersection
+                    addToItem(upperNestingSelectors as SelectorList);
```

Comment:
this can be simplified to ignore empty/irrelevant parent scopes, if you've reached the start with nothing interesting there, revert back to root context.

Reply:
You are 100% correct. Running back on a selector might encounter a nesting selector that should be inferred and reaching the end ignores the upper nesting context and resets to the stylesheet context (root).

This simplified so much of the code!